### ISSUE TITLE
An API query is needed to return or query a list of all "Interests" from the owners/profiles tables.

### ISSUE THREAD

**Issue by emilyashley** [MapStory/mapstory#701]
I've submitted interests (turned into keywords see #700 ) on a profile edit form.

When searching on the Explore storyteller tab these interests I submitted are no longer available to populate autocomplete

<img width="890" alt="screen shot 2017-05-12 at 4 36 58 pm" src="https://cloud.githubusercontent.com/assets/15912063/26017888/83f2d4ea-3731-11e7-9825-cf9bde3f2ba4.png">
<img width="824" alt="screen shot 2017-05-12 at 4 36 32 pm" src="https://cloud.githubusercontent.com/assets/15912063/26017890/873aa6f0-3731-11e7-9d5c-af9c83caddbc.png">

Check to see if these entries are making it into the keywords/api and if they are getting associated with user's profiles. 

note: the request I'm using for autocomplete that was working before the transition to core geonode is 
```
$http.get(KEYWORDS_ENDPOINT,  {params: {slug__icontains: searchText} || {}})
           .then(function(response){ 
              return response.data.objects;
          });   
```
 with `searchText` being the text from the input field

**Comment by sarasafavi** [comment]
So the problem here is centered around GeoNode 2.6's major change to using "h_keywords" (hierarchical keywords); the `KEYWORDS_ENDPOINT` in the example above returns results from the h_keyword list, but not non-hierarchical keywords. Currently our profiles save keywords as "tags", not h_keywords _(forgive the jargon: TL;DR, two different types of searchable keywords fields)_

So, there are a couple ways to work this. 

We can easily update our profiles to save h_keywords, which would give the same behavior as old MapStory. _However_ that old behavior results in the following workflow:

Explore --> StoryTellers
_enter value in 'INTEREST' form field_
Autocomplete results list _all_ keywords in MapStory: including Storylayer & Story keywords, as well as StoryTeller Interests 

If we want, while we're fixing this we can go ahead update workflow to the following:

Explore --> StoryTellers
_enter value in 'INTEREST' form field_
Autocomplete results list _only_ StoryTeller Interests

@emilyashley - thoughts on optimal UX here (changing known behavior vs more intuitive behavior)? We can discuss further next week when we're both back on task, so no rush.

**Comment by emilyashley** [comment]
Thanks, @sarasafavi. (For the record we went with the Autocomplete results list _only_ for StoryTeller Interests route)

 With #812 merged in and #700 closed, I should have everything In need to re-include them and populate the autocomplete. Assigning myself to the remaining bits of this issue now.

**Comment by emilyashley** [comment]
I've turned off the autocomplete behavior in the UI (via #1032 ) until I know the best way to search for "Interests" as they are no longer queryable from the Keywords API and live solely in the profiles/mapstoryprofiles tables.
